src/parse_map.c

```diff
@@ -0,0 +1,95 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   parse_map.c                                        :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: abasdere <abasdere@student.42.fr>          +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2024/03/07 13:03:57 by abasdere          #+#    #+#             */
+/*   Updated: 2024/03/08 14:04:33 by abasdere         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "parsing.h"
+
+#define VALID_CHAR "10NSEW "
+#define PLAYER_VIEW "NSEW"
+#define CHAR_ERROR "Invalid char found in the map"
+#define NO_PLAYER "No starting position was found"
+#define MULTIPLE_PLAYER "Multiple starting position found"
+#define MALLOC_ERROR "Malloc failed"
+
+static void	create_rectangle(t_list *line, t_map *map, t_collector *collector)
+{
+	char	*tmp;
+	size_t	i;
+	size_t	len;
+
+	i = 0;
+	map->map = ccalloc(map->heigh + 1, sizeof(char *), collector);
+	while (line)
+	{
+		len = ft_strlen((char *)line->content);
+		if (len < map->width)
+		{
+			tmp = ft_calloc(map->width + 1, sizeof(char));
+			if (!tmp)
+				cerror(MALLOC_ERROR, collector);
+			ft_memcpy(tmp, (char *)line->content, len);
+			(free(line->content), line->content = tmp);
+		}
+		len = -1;
+		while (((char *)line->content)[++len])
+			if (((char *)line->content)[len] == ' ')
+				((char *)line->content)[len] = '\0';
+		map->map[i++] = line->content;
+		line = line->next;
+	}
+}
+
+static int	parse_player(size_t i, size_t j, t_map *map, const char *line)
+{
+	if (map->player.pos.x != 0 && map->player.pos.y != 0)
+		return (1);
+	map->player = (t_player){{i, j}, {0, 0}};
```

Comment:
may create errors if stack allocation is freed

Reply:
how so ? map's allocation comes from main